### ISSUE TITLE
Context-aware D not learning

### ISSUE THREAD

**Issue by BianchiGiulia** [RUCAIBox/RecBole#2046]
Hello!
I'm trying to

**Comment by mmosc** [comment]
Hi!

Maybe I can help here: have you tried setting `freeze=False`?

From pytorch `from_pretrained` [documentation](https://pytorch.org/docs/stable/generated/torch.nn.Embedding.html#torch.nn.Embedding.from_pretrained):
 `freeze` ([bool](https://docs.python.org/3/library/functions.html#bool), optional) – If `True`, the tensor does not get updated in the learning process. Equivalent to `embedding.weight.requires_grad = False`. Default: `True`

**Comment by BianchiGiulia** [comment]
Hey! Thank you fro your prompt reply!

I thought that any context-aware algorithm could just use any pre-trained embedding as described in the [doc](https://recbole.io/docs/user_guide/usage/load_pretrained_embedding.html), but a quick experiment with and without those embedding gave the exact same metric results, so I guess you're right and I should dive deeper into the OG code😅, or just choose for a different algorithm. 

Thank you so much!!
